### ISSUE TITLE
UK VISA 3DS requestPaymentMethod never triggers required authentication

### ISSUE THREAD

**Issue by ChrisHSandN** [braintree/braintree-web-drop-in#807]
### General information

* SDK/Library version: 1.33.0
* Environment: Production
* Browser and OS: Chrome latest Windows 10

### Issue description
We are experiencing almost complete failure with VISA cards due to 3DS authentication issues. This seems to be coinciding with the deadline of 14th March 2022 for UK SCA compliance.

We are able to reliably replicate the issue using a card under our control. 
1. The drop-in accepts the card number
2. Client side, we request a 3DS check using `requestPaymentMethod(threeDSecure: {threeDSecureParameters...})`
3. This comes back, with no errors, without showing the 3DS box and as `liabilityShifted:true`.
4. Server side we convert the nonce to a PaymentMethod, attempt to charge it and get `2099 "Authentication Required"`
  > The customer's bank declined the transaction because a 3D Secure authentication was not performed during checkout. Have the customer authenticate using 3D Secure, then attempt the authorization again.

#### Console output from using `cardinalSDKConfig.logging.level ='verbose'`
```
Base.Main: Using Songbird.js v1.35.0
instrument.ts:129 Base.Events.SetupHandler: Setting up Cardinal.js.
instrument.ts:129 Base.Events.SetupHandler: JWT Detected on Setup. Processing will continue using JWT with signature [X5XACC9fsf7L8Je97M41oNtzpQzykWJVlGgjQ2DmnZg]
instrument.ts:129 Base.Message.Init: Processing Init message...
instrument.ts:129 Base.SupportedPaymentsUtility: Missing required field for Hosted Fields: AccountNumber
instrument.ts:129 Base.cors: Sending async post request to https://centinelapi.cardinalcommerce.com/V1/Order/JWT/Init with a timeout value of 6000ms
instrument.ts:129 Base.cors: CORS message completed successfully
instrument.ts:129 Base.Payment.ConsumerAuthentication: Initializing plugin
instrument.ts:129 Base.UiManager: Loading UI Framework [cardinal]
instrument.ts:129 Base.Payment.ConsumerAuthentication: CCA Initialized
instrument.ts:129 Base.cors: Sending async post request to https://writer.cardinalcommerce.com/prod/log with a timeout value of 6000ms
instrument.ts:129 Base.Message.Init: All expected payments have finished loading.
instrument.ts:129 Base.cors: CORS message completed successfully
instrument.ts:129 Base.Message.BinEnabled: Processing DeviceFingerprinter Bin Enabled message...
instrument.ts:129 Base.cors: Sending async post request to https://geo.cardinalcommerce.com/DeviceFingerprintWeb/V2/Bin/Enabled with a timeout value of 6000ms
instrument.ts:129 Base.cors: CORS message completed successfully
instrument.ts:129 Base.cors: Sending async post request to https://writer.cardinalcommerce.com/prod/log with a timeout value of 6000ms
instrument.ts:129 Base.cors: CORS message completed successfully
```

#### Cardinal response
```
{
  "paymentMethod": {
    "type": "CreditCard",
    "nonce": "f5c8d83d-4ad6-1af1-811b-49f00cb4b950",
    "description": "ending in 22",
    "isLocked": false,
    "securityQuestions": [
      "cvv"
    ],
    "threeDSecureInfo": {
      "liabilityShifted": true,
      "liabilityShiftPossible": true,
      "status": "authenticate_successful",
      "enrolled": "Y",
      "cavv": "AJkBBEVHJQAAABEmgmBpdQAAAAA=",
      "xid": null,
      "acsTransactionId": "cc78d8f7-58c6-4a14-b531-7886e1bdd073",
      "dsTransactionId": "f60b8609-efab-4ff3-9309-0dbcf30d2f65",
      "eciFlag": "05",
      "acsUrl": null,
      "paresStatus": "Y",
      "threeDSecureAuthenticationId": "s4tvr44t8kdjjdkx5k",
      "threeDSecureServerTransactionId": "eecdaf55-5379-4eed-aca2-ce2ceb9b43e6",
      "threeDSecureVersion": "2.2.0",
      "lookup": {
        "transStatus": "Y",
        "transStatusReason": null
      },
      "authentication": {
        "transStatus": null,
        "transStatusReason": null
      }
    },
    "consumed": false,
    "default": true,
    "hasSubscription": false,
    "details": {
      "cardType": "Visa",
      "lastTwo": "22",
      "lastFour": "[redacted]",
      "bin": "[redacted]",
      "cardholderName": null,
      "expirationYear": "[redacted]",
      "expirationMonth": "[redacted]"
    },
    "binData": {
      "prepaid": "No",
      "healthcare": "No",
      "debit": "Yes",
      "durbinRegulated": "No",
      "commercial": "Yes",
      "payroll": "No",
      "issuingBank": "BARCLAYS BANK UK PLC",
      "countryOfIssuance": "GBR",
      "productId": "G"
    }
  },
  "threeDSecureInfo": {
    "liabilityShifted": true,
    "liabilityShiftPossible": true
  },
  "lookup": {
    "acsUrl": null,
    "md": "s4tvr44t8kdjjdkx5k",
    "termUrl": "[redacted]",
    "pareq": null,
    "threeDSecureVersion": "2.2.0",
    "transactionId": "NjJZ0Vkel3A9KxhhZkQ0"
  }
}
```

### Support
As a further point, we have tried to resolve this via Braintree support but have been stymied at every turn: 
1. We have opened a ticket with Braintee support but have not received any escalation to technical support since late February.
2. We used to be able to phone Braintree for technical support; however the numbers have now been removed and replaced with only a sales number. Upon phoning that we were informed that the technical number has been removed due to “the volume of calls”!
3. We have directly contacted CardinalCommerce, passing them the verbose console output of their SDK and the relevant API calls. Initially it was promising; they were technical, helpful and had a working phone number. But then someone at Braintree must have realised our sneaky plan to sidestep them and they changed their tune and stated that “Braintree has asked that if we receive inbounds from their merchants to have you open a support ticket with them” (see point 1 and 2).

**Comment by ChrisHSandN** [comment]
Closing, as we were able to resolve the issue with help from support 👍

Posting this in case anyone else is suffering from 3DS related issues:

The core of the problem turns out to be that almost everything to do with PaymentMethod tokens (note _token_ and not _nonce_) are incompatible with Braintree’s 3DS2 implementation. Even with the whole of the EU now mandated to use SCA, the bulk of the current API documentation still continues to show the main examples as first converting a nonce to a token, then using these to identify a PaymentMethod.

What the drop-in returns (once 3DS is enabled) is now known as a “3DS enriched nonce”. You get one shot to use this – either in a `Transaction::sale()` or `Subscription::create()`. Do not use this in a `PaymentMethod::create()` call or the 3DS enrichment will be (silently) discarded.
